docker/rx-mission/app.py

```diff
@@ -0,0 +1,19 @@
+import json
+
+from flask import Flask, jsonify
+
+app = Flask(__name__)
+
+
+@app.route("/product/<string:product_id>", methods=["GET"])
+def handle_request(product_id):
+    with open("./fixtures/rx-mission-products.json", encoding="utf-8") as json_file:
+        parsed_json = json.load(json_file)
+
+    for product in parsed_json["products"]:
+        if product["id"] == product_id:
+            return jsonify(product)
```

Comment:
if you parse and process the objects once you can use a simple hash map lookup:

```py
# module level
products = {product["id"]: product for product in parsed_json["products"]}

# in request handler
product = products[product_id]
return jsonify(product)
```

Reply:
Yeah that's a bit nicer. I've changed it